models/Discord/Commands/DiscordCommandWhoIs.js

```diff
@@ -10,15 +10,21 @@ class DiscordCommandWhoIs extends DiscordCommand {
   onRun(message, permissions, args) {
     var config = this.subsystem.manager.getSubsystem("Config").config;
     if(args.length < 1) {
-      message.reply("Usage is `" + config.discord_command_character + "whois [@UserName`");
+      message.reply("Usage is `" + config.discord_command_character + "whois [@UserName/Ckey]`");
       return;
     }
     var user = undefined;
+    var provided_ckey = args.join("");
+    var punctuation = [".", ",", "-", "_", ";", ":"];
+    for(var i = 0; i< punctuation.length; i++) {
+      provided_ckey = provided_ckey.split(punctuation[i]).join("");
+    }
```

Comment:
you forgot to convert to lowercase.

Also this is a better way to convert to ckey: 
```js
args.join("").toLowerCase().replace(/[^a-z0-9@]/gi, "")
```
which is the method used in other parts of this codebase.

Reply:
I just stole the code from the !notes command honk